tests/integration-tests/tests/common/schedulers_common.py

```diff
@@ -382,13 +382,15 @@ def assert_job_state(self, job_id, expected_state):  # noqa: D102
     def compute_nodes_count(self, filter_by_partition=None):  # noqa: D102
         return len(self.get_compute_nodes(filter_by_partition))
 
-    def get_compute_nodes(self, filter_by_partition=None):  # noqa: D102
+    def get_compute_nodes(self, filter_by_partition=None, all_nodes=None):  # noqa: D102
         command = "sinfo --Node --noheader --responding"
         if filter_by_partition:
             command += " --partition {}".format(filter_by_partition)
         # Print first and fourth columns to get nodename and state only (default partition contains *)
         # Filter out nodes that are not responding or in power saving states
-        command += " | awk '{print $1, $4}' | grep -v '[*#~%]' | awk '{print $1}'"
+        command += (
+            " | awk '{print $1, $4}' | grep -v '[*#~%]' | awk '{print $1}'" if not all_nodes else " | awk '{print $1}'"
+        )
```

Comment:
I believe we should also move the `--responding` flag to the if condition in line 392: otherwise the case with `all_nodes` will still miss the non-responding nodes.

Out of scope: we should rewrite this whole method (by default this implementation doesn't return all nodes in a cluster, only the responding ones, and I am afraid this is not consistent with the definition in the parent class).

Reply:
removed